### PR TITLE
[RepositoryProvider] Add editable to checkoutClone

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Depends on #681

File: Sources/SourceControl/GitRepository.swift [hunk 1]
```diff
@@ -71,24 +71,42 @@ public class GitRepositoryProvider: RepositoryProvider {
...
+            // For editable clones, i.e. the user is expected to directly work on them, first we create
+            // a clone from our cache of repositories and then we replace the remote to the one originally
+            // present in the bare repository.
+            try system(
+                    Git.tool, "clone", sourcePath.asString, destinationPath.asString, message: nil)
+            // The default name of the remote.
+            let origin = "origin"
+            // Get the remote from original repo.
+            let source = GitRepository(path: sourcePath)
+            guard let remote = try source.remotes().first, remote.name == origin else {
```
Comment (ddunbar) [inline Sources/SourceControl/GitRepository.swift]: We probably shouldn't assume the origin is first, and should instead search for it.

Comment (ddunbar) [inline Sources/SourceControl/GitRepository.swift]: It feels a little unfortunate to _derive_ the source origin here, given that the client actually knows what it should be, but maybe it isn't worth plumbing that through?

Comment (aciidb0mb3r) [inline Sources/SourceControl/GitRepository.swift]: made the searching change but need to decide if we should directly inject remote. Landing this for now

Comment (ddunbar) [inline Sources/SourceControl/GitRepository.swift]: where did we land on this point?

Comment (aciidb0mb3r) [inline Sources/SourceControl/GitRepository.swift]: will change to use remote from input

**Pull request merged**